### PR TITLE
Razorwire nerf

### PR DESCRIPTION
**Pull request by jroinc**
## About The Pull Request

Grenades and canisters are left unchanged.

The minimum chem requirements for a razorburn mix that does anything is 7 nanites and 3 foam. Breaking this down, there is a 3/3 hardcap (smaller amounts will mix but won't do anything), and an additional 7u nanites catalyst softcap. 

Razorburn no longer deletes reagents in it's holder when it reacts, so it's viable to just directly add foaming agent to the nanite beaker (and is probably the easiest way to do so), but all razorburn reactions that aren't exactly 7/7 with no leftover nanites will *actually* have nanites in the foam now, as intended.

## Why It's Good For The Game

0-cost infini-fields of razorwire are unhealthy for the game. Minimize without removing, yay.

## Changelog
:cl:

balance: razorwire thing
/:cl:

### CODE REVIEW

**Comment by Luxyboo** [comment]
FUCK RAZORWIRE

File: code/game/objects/effects/effect_system/foam.dm [hunk 1]
```diff
@@ -144,7 +144,7 @@
 	else
 		src.location = get_turf(location)
 
-	src.spread_amount = round(sqrt(spread_amount / 3), 1)
+	src.spread_amount = round(sqrt(spread_amount / 3))
 	carry.copy_to(carrying_reagents, carry.total_volume)
 	src.foam_flags = foam_flags
```
Comment (TiviPlus) [inline code/game/objects/effects/effect_system/foam.dm:149]: use FLOOR?

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/effect_system/foam.dm (hunk 1 above)
Comment (jroinc) [reply code/game/objects/effects/effect_system/foam.dm:149]: How?

Couldn't find it in the documentation?

**Review by BraveMole — CHANGES_REQUESTED** [review]
Fix your cl, say that all micro-reaction (including razor wire) were nuked.

Maybe a warning when the final spread amount is lower than one (The reaction failed because of a lack of reagants or smth). Although it may be hard to code, so if you don't find a soution then whatever

File: code/game/objects/effects/effect_system/foam.dm (hunk 1 above)
Comment (BraveMole) [reply code/game/objects/effects/effect_system/foam.dm:149]: Floor does exactly the same thing tivi. And it's not in the doc jpr, it's a custom macro

**Comment by jroinc** [comment]
> 
> 
> Fix your cl, say that all micro-reaction (including razor wire) were nuked.
> 
> Maybe a warning when the final spread amount is lower than one (The reaction failed because of a lack of reagants or smth). Although it may be hard to code, so if you don't find a soution then whatever

I'm not sure *why* it's doing that. I expected a result of 0 to have zero spread (the tile it spawned on and nothing else). If you see *why* it's doing that, please let me know, cause I don't see why.

**Review by BraveMole — CHANGES_REQUESTED** [review]
image

try to add a check of start, if spread is equal to zero do nothing. Check that this doesn't break anything (like are the reagants still consummed), because i have no clue

**Comment by jroinc** [comment]
> image
> 
> try to add a check of start, if spread is equal to zero do nothing. Check that this doesn't break anything (like are the reagants still consummed), because i have no clue

Eh, it's been two weeks with this sitting here while other shit gets speedmerged. I don't care about these anymore. If you wanna add it, add it, otherwise close it. I'll close this and any of my unadded PR's in a few days regardless, add any you actually want.

**Comment by BraveMole** [comment]
> Eh, it's been two weeks with this sitting here while other shit gets speedmerged. I don't care about these anymore. If you wanna add it, add it, otherwise close it. I'll close this and any of my unadded PR's in a few days regardless, add any you actually want.

I alrady pinged you how to fix. All of your other prs are either not compiling, have broken linters, have unanswered request.

This one is creating foam that does nothing, which leads to player confusion ; if you don't want to fix this, fine, but then this pr is not coming in. And i'll just nuke chem razor wire like i said i would

**Pull request closed without merging**